packages/components/select/src/useSelect.ts

```diff
@@ -210,6 +210,15 @@ export const useSelect = (props, states: States, ctx) => {
     () => props.placeholder,
     (val) => {
       states.cachedPlaceHolder = states.currentPlaceholder = val
+
+      const hasValue =
+        props.multiple &&
+        Array.isArray(props.modelValue) &&
+        props.modelValue.length > 0
+
+      if (hasValue) {
+        states.currentPlaceholder = ''
+      }
```

Comment:
It seems to be consistent with the logic of the following code, maybe it can be extracted
https://github.com/element-plus/element-plus/blob/2ab408ae4856afe4a64d33d4ddde52cbc5c85a61/packages/components/select/src/select.vue#L547-L552